samcli/lib/build/app_builder.py

```diff
@@ -208,39 +217,52 @@ def update_template(template_dict: Dict, original_template_path: str, built_arti
 
         for logical_id, resource in template_dict.get("Resources", {}).items():
 
-            if logical_id not in built_artifacts:
-                # this resource was not built. So skip it
+            full_path = get_full_path(stack_path, logical_id)
+            is_artifact = full_path in built_artifacts
```

Comment:
what does `is_artifact` denote? Does this mean that stack is built and artifacts are present? If yes then in what case it won't be present?

Reply:
let's change to `has_build_artifact`. It indicates whether "this resource" has a artifacts generated by sam build.